plugins/filter_aws/aws.c

```diff
@@ -390,7 +390,7 @@ static int get_ec2_tag_values(struct flb_filter_aws *ctx)
         /* fetch tag value using path: /latest/meta-data/tags/instance/{tag_name} */
         tag_value_path_len = ctx->tag_keys_len[i] + 1 +
                              strlen(FLB_AWS_IMDS_INSTANCE_TAG);
-        tag_value_path = flb_sds_create_size(tag_value_path_len);
+        tag_value_path = flb_sds_create_size(tag_value_path_len + 1);
```

Comment:
@mwarzynski I don't understand this... it looks like you already have a +1 for the "/" char in the tag_value_path_len above? 

(If you need an extra char, it should be plus 2 up there I think)

Why does this only happen with very long tag names??

Also, flb_sds_printf will re-alloc if needed according to my understanding: https://github.com/fluent/fluent-bit/blob/master/src/flb_sds.c#L437

Reply:
@PettitWesley I don't understand this fully either. Based on intuition, I assumed it's related to \0 and added +1 to the required length.

Reply:
To be honest, I didn't have a lot of time to debug this properly and just wanted to push a working fix. It would be a pitty if we would release ec2 tags feature without this fix. As I recall we suspect the release is supposed to happen sooner than later.

Reply:
It worked for the tag names with length < 32 because of the `if (len < 64) len = 64;` in https://github.com/fluent/fluent-bit/blob/master/src/flb_sds.c#L417

For tag names with length >= 32, `flb_sds_printf` appears to truncate the output string to the size of the provided char array. There is no way to tell if output string was truncated or not as we get `!= NULL`.

Fortunately this is the only place where I used the printf functionality, so nothing else is affected. (Not sure about other parts of the codebase though.)

Reply:
@mwarzynski https://github.com/fluent/fluent-bit/issues/7143